internal/vm/vsock.go

```diff
@@ -254,11 +254,16 @@ func tryAccept(logger *logrus.Entry, listener net.Listener, port uint32) (net.Co
 // It's only intended to be used for connect/ack messages, not general purpose reads
 // after the vsock connection is established fully.
 func tryConnRead(conn net.Conn, expectedRead string, timeout time.Duration) error {
-	conn.SetDeadline(time.Now().Add(timeout))
-	defer conn.SetDeadline(time.Time{})
+	err := conn.SetDeadline(time.Now().Add(timeout))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = conn.SetDeadline(time.Time{})
+	}()
```

Comment:
It appears `SetDeadline` probably only fails if the socket itself is invalid (in which case we'll just get errors when trying to do the read/write), but it's still good to check these for extra safety. But in that case we should also return an error if the `SetDeadline` call in the defer fails. If that did fail but the socket was somehow still usable, then the caller would be getting a socket with an unexpected deadline.

This may be an okay use case for using a named return, but we'll also need to make sure we don't overwrite any other error already being returned.

Reply:
I somehow thought that inside defer is too late to return `err`. Will update.